src/libponyc/pass/docgen.c

```diff
@@ -755,6 +789,145 @@ static void doc_methods(docgen_t* docgen, docgen_opt_t* docgen_opt,
     doc_method(docgen, docgen_opt, p->ast);
 }
 
+static char* concat(const char *s1, const char *s2)
+{
+  size_t str_size = strlen(s1) + strlen(s2) + 1;
+  char* result = (char*) malloc(str_size); //+1 for the null-terminator
+  strcpy(result, s1);
+  strcat(result, s2);
+  result[str_size - 1] = '\0';
+  return result;
+}
+
+static doc_sources_t* copy_source_to_doc_src(docgen_t* docgen, source_t* source, const char* package_name)
+{
+  pony_assert(docgen != NULL);
+  pony_assert(source != NULL);
+  pony_assert(package_name != NULL);
+
+  doc_sources_t* result = (doc_sources_t*) calloc(1, sizeof(doc_sources_t));
+
+  char filename_copy[FILENAME_MAX];
+  strcpy(filename_copy, source->file);
+
+  const char* just_filename = get_file_name(filename_copy);
+  char* filename = (char*) calloc(strlen(just_filename) + 1, sizeof(char));
+  strcpy(filename, just_filename);
+  const char* filename_without_ext = remove_ext(filename, '.', 0);
+  const char* filename_md_extension = concat(filename_without_ext, ".md");
+
+  // Absolute path where a copy of the source will be put.
+  char source_dir[FILENAME_MAX];
+  path_cat(docgen->doc_source_dir, package_name, source_dir);
+
+  //Create directory for [documentationDir]/src/[package_name]
+  pony_mkdir(source_dir);
+  
+  // Get absolute path for [documentationDir]/src/[package_name]/[filename].md
+  char* path = (char*) malloc(sizeof(char) * FILENAME_MAX);
+  path_cat(source_dir, filename_md_extension, path);
+
+  // Get relative path for [documentationDir]/src/[package_name]/
+  // so it can be written in the mkdocs.yml file.
+  const char* doc_source_dir_relative = concat("src/", package_name);
+  const char* old_ptr = doc_source_dir_relative;
+  doc_source_dir_relative = concat(doc_source_dir_relative, "/");
+  free((void*) old_ptr);
+  
+  // Get relative path for [documentationDir]/src/[package_name]/[filename].md
+  const char* doc_path = concat(doc_source_dir_relative, filename_md_extension);
+
+  free((void*) filename_without_ext);
+  free((void*) filename_md_extension);
+  free((void*) doc_source_dir_relative);
+
+  // Section to copy source file to [documentationDir]/src/[package_name]/[filename].md
+  FILE* file = fopen(path, "w");
+
+  if (file != NULL) {
+    // Adding a 'special' header so it can be identified 
+    // as a complete Pony source file by JavaScript to add line numbers.
+    fprintf(file, "<div class=\"pony-full-source\" hidden>\n");
```

Comment:
sorry, i have been misleading, the `hidden` attribute needs to be removed.

Reply:
Done. By the way, there is still the embedded extra.js and extra.css that should be deleted because they will part of your theme (as well as the method that copy them into the generated documentation directory). I leave them for now and will delete once your work is ready.